cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -332,92 +82,60 @@ def import_status_handler(request, course_key_string, filename=None):
-    if 'application/x-tgz' in requested_format:
-        name = course_module.url_name
-        export_file = NamedTemporaryFile(prefix=name + '.', suffix=".tar.gz")
-        root_dir = path(mkdtemp())
-
-        try:
-            export_to_xml(modulestore(), contentstore(), course_module.id, root_dir, name)
-
-            logging.debug(u'tar file being generated at {0}'.format(export_file.name))
-            with tarfile.open(name=export_file.name, mode='w:gz') as tar_file:
-                tar_file.add(root_dir / name, arcname=name)
-        except SerializationError as exc:
-            log.exception(u'There was an error exporting course %s', course_module.id)
-            unit = None
-            failed_item = None
-            parent = None
-            try:
-                failed_item = modulestore().get_item(exc.location)
-                parent_loc = modulestore().get_parent_location(failed_item.location)
-
-                if parent_loc is not None:
-                    parent = modulestore().get_item(parent_loc)
-                    if parent.location.category == 'vertical':
-                        unit = parent
-            except:  # pylint: disable=bare-except
-                # if we have a nested exception, then we'll show the more generic error message
-                pass
-
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'raw_err_msg': str(exc),
-                'failed_module': failed_item,
-                'unit': unit,
-                'edit_unit_url': reverse_usage_url("container_handler", parent.location) if parent else "",
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        except Exception as exc:
-            log.exception('There was an error exporting course %s', course_module.id)
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'unit': None,
-                'raw_err_msg': str(exc),
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        finally:
-            shutil.rmtree(root_dir / name)
-
-        wrapper = FileWrapper(export_file)
-        response = HttpResponse(wrapper, content_type='application/x-tgz')
-        response['Content-Disposition'] = 'attachment; filename=%s' % os.path.basename(export_file.name.encode('utf-8'))
-        response['Content-Length'] = os.path.getsize(export_file.name)
-        return response
+    export_url = reverse(
+        "course_import_export_handler",
+        kwargs={
+            "course_key_string": unicode(course_key),
+        }
+    ) + '?accept=application/x-tgz'
+
+    export_url += "&{0}".format(
```

Comment:
Nitpick: rather than doing this with two different string concatenations, it might be clearer and more extensible to use a map. Something like this:
```python
queryargs = {
    "accept": "application/x-tgz",
    "redirect": reverse_course_url(
        "export_handler", unicode(course_key)
    ),
}
querystring = "&".join(
    "{key}={value}".format(key=urlencode(key), value=urlencode(value))
    for key, value in queryargs.items()
)
export_url += "?" + querystring
```